File: src/main/java/net/scriptgate/conway/field/Cell.java
package net.scriptgate.conway.field;

import net.scriptgate.conway.common.Point2D;

public class Cell {

    final Point2D position;
    final boolean alive;

    Cell(int x, int y, boolean alive) {
        this.position = new Point2D(x, y);
        this.alive = alive;
    }

    public static Cell createLiveCell(int x, int y) {
        return new Cell(x, y, true);
    }

    public static Cell createDeadCell(int x, int y) {
        return new Cell(x, y, false);
    }

}

File: src/main/kotlin/net/scriptgate/conway/field/Cell.kt
package net.scriptgate.conway.field

import net.scriptgate.conway.common.Point2D

class Cell internal constructor(x: Int, y: Int, internal val alive: Boolean) {

    internal val position: Point2D = Point2D(x, y)

    fun translate(x: Int, y: Int): Cell {
        return Cell(position.x + x, position.y + y, alive)
    }

    companion object {
        fun createLiveCell(x: Int, y: Int): Cell {
            return Cell(x, y, true)
        }
    }

}
